pkg/processor/runtime/python/py/_nuclio_wrapper_as_server.py

```diff
@@ -0,0 +1,290 @@
+        server_sock.setblocking(False)
+
+        self._logger.info(f"Server started on {self._host}:{self._port}, max_connections: {self._max_connections}")
+
+        # Register server socket with epoll
+        self.selector.register(server_sock, selectors.EVENT_READ)
+        self.connections[server_sock.fileno()] = server_sock
+        try:
+            while True:
+                events = await self.select_events()
+                # Now awaiting the select operation
+                for key, event in events:
+                    sock = key.fileobj
+                    if sock == server_sock:
+                        # Accept new connection
+                        client_sock, addr = server_sock.accept()
+                        self._logger.info(f"Accepted connection from {addr}")
+                        client_sock.setblocking(False)
+                        self.connections[client_sock.fileno()] = client_sock
+                        task = self._loop.create_task(self._process_connection(client_sock))
+                        self.tasks[client_sock.fileno()] = task
+                    else:
+                        self._logger.error(f"Unexpected event: {event}")
+                await asyncio.sleep(0.2)
+        except KeyboardInterrupt:
+            self._logger.info("Shutting down server")
+        finally:
+            self._logger.info("Closing all connections")
+            for sock in self.connections.values():
+                sock.close()
+
+    async def select_events(self, timeout=0.1):
+        # Run the selector.select in the default executor (this is a blocking operation)
+        events = await self._loop.run_in_executor(None, self.selector.select, timeout)
+        return events
+
+    async def initialize(self):
+
+        # call init_context
+        await self._initialize_context()
+
+        # register to the SIGUSR1 and SIGUSR2 signals, used to signal termination/draining respectively
+        self._register_to_signal()
+
+        await self._send_data_on_control_socket({
+            'kind': 'wrapperInitialized',
+            'attributes': {'ready': 'true'}
+        })
+
+    async def _stop_processing(self):
+        self.shutdown_event.set()
+
+        # Cancel all active tasks
+        for fileno, task in list(self.tasks.items()):
+            self._logger.info(f"Cancelling task for connection {fileno}")
+            task.cancel()
+        # Close all active sockets
+        for fileno, sock in list(self.connections.items()):
+            self._logger.info(f"Closing connection {fileno}")
+            sock.close()
+        self.selector.close()
+
+    async def _shutdown(self, error_code=0):
+        """Shutdown the server and cleanup resources."""
+        self._logger.info("Shutting down...")
+        try:
+            await self._stop_processing()
+        finally:
+            super()._shutdown()
+
+
+def parse_args():
+    parser = get_parser_with_common_args()
+    return parser.parse_args()
```

Comment:
If there aren't custom args, this whole function can move to `wrapper_common.py`

Reply:
@TomerShor there might be custom args, that's an interface for future